unity-renderer/Assets/DCLServices/WearablesCatalogService/LambdasWearablesCatalogService.cs

```diff
@@ -0,0 +1,285 @@
+        private readonly Dictionary<string, UniTaskCompletionSource<WearableItem>> awaitingWearableTasks = new ();
+        private readonly List<string> pendingWearablesToRequest = new ();
+
+        public LambdasWearablesCatalogService(BaseDictionary<string, WearableItem> wearablesCatalog)
+        {
+            WearablesCatalog = wearablesCatalog;
+        }
+
+        public void Initialize()
+        {
+            serviceCts = serviceCts.SafeRestart();
+
+            // All the requests happened during the same frames interval are sent together
+            CheckForSendingPendingRequestsAsync(serviceCts.Token).Forget();
+
+            // Check unused wearables (to be removed from our catalog) only every [TIME_TO_CHECK_FOR_UNUSED_WEARABLES] seconds
+            CheckForUnusedWearablesAsync(serviceCts.Token).Forget();
+        }
+
+        public void Dispose()
+        {
+            serviceCts.SafeCancelAndDispose();
+            serviceCts = null;
+            Clear();
+        }
+
+        public async UniTask<IReadOnlyList<WearableItem>> RequestOwnedWearablesAsync(string userId, int pageNumber, int pageSize, CancellationToken ct)
+        {
+            if (!ownerWearablesPagePointers.TryGetValue(userId, out var pagePointer))
+            {
+                ownerWearablesPagePointers[userId] = pagePointer = new (
+                    WEARABLES_BY_OWNER_END_POINT.Replace("{userId}", userId),
+                    pageSize, ct, this);
+            }
+
+            var pageResponse = await pagePointer.GetPageAsync(pageNumber, ct);
+            AddWearablesToCatalog(pageResponse.response.wearables);
+
+            return pageResponse.response.wearables;
+        }
+
+        public async UniTask<IReadOnlyList<WearableItem>> RequestBaseWearablesAsync(CancellationToken ct)
+        {
+            if (!wearablesByCollectionPagePointers.TryGetValue(BASE_WEARABLES_COLLECTION_ID, out var pagePointer))
+            {
+                wearablesByCollectionPagePointers[BASE_WEARABLES_COLLECTION_ID] = pagePointer = new (
+                    WEARABLES_BY_COLLECTION_END_POINT.Replace("{collectionId}", BASE_WEARABLES_COLLECTION_ID),
+                    1, ct, this);
+            }
+
+            var pageResponse = await pagePointer.GetPageAsync(1, ct);
+            AddWearablesToCatalog(pageResponse.response.wearables);
+
+            return pageResponse.response.wearables;
+        }
+
+        public async UniTask<IReadOnlyList<WearableItem>> RequestThirdPartyWearablesByCollectionAsync(string userId, string collectionId, CancellationToken ct)
+        {
+            if (!thirdPartyCollectionPagePointers.TryGetValue((userId, collectionId), out var pagePointer))
+            {
+                thirdPartyCollectionPagePointers[(userId, collectionId)] = pagePointer = new (
+                    WEARABLES_BY_THIRD_PARTY_COLLECTION_END_POINT
+                       .Replace($"{userId}", userId)
```

Comment:
careful here, you are trying to replace the value of user id. You should remove the `$` to replace `"{userId}"` instead

Reply:
Good catch! I didn't relalize, thanks! Fixed in: https://github.com/decentraland/unity-renderer/pull/4084/commits/4077bbd1967325a7296de18bbf679dfa315bd344